src/Microsoft.IdentityModel.Tokens/TokenValidationResult.cs

```diff
@@ -56,7 +70,14 @@ internal TokenValidationResult(SecurityToken securityToken, TokenHandler tokenHa
                 if (!_hasIsValidOrExceptionBeenRead)
                     LogHelper.LogWarning(LogMessages.IDX10109);
 
-                return _claims.Value;
+                IDictionary<string, object> claims = _claims;
+                if (claims is null && ClaimsIdentity is { } ci)
```

Comment:
I don't understand why we need the assignment to local variable claims or the need to call ClaimsIdentity.